RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,174 @@
+
+To fulfil the requirements of RFC003 this RFC defines:
+
+- The *identity* to be used when negotiating a SWAP on the Bitcoin ledger.
+- How to construct a Hash Time Lock Contract (HTLC) to lock the Bitcoin asset on the Bitcoin blockchain.
+- How to deploy, redeem and refund the HTLC during the execution phase of RFC003
+
+
+## Motivation
+
+The motivation for creating this RFC is to allow implementations to negotiate and execute [RFC003 Basic HTLC Atomic Swaps](./RFC-003-SWAP-basic.md) using the COMIT protocol. The identity definition introduced in this RFC should also be useful for future protocols.
+
+## Bitcoin Identity
+
+The Identity to be used on Bitcoin is the 20-byte *pubkeyhash* which is the result of applying SHA-256 and then RIPEMD-160 to a user's compressed SECP256k1 public key.
+The compressed public key is used because it needs to be compatiable with segwit transactions (see [BIP143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Restrictions_on_public_key_type)).
+
+While it may seem more intuitive to use a Bitcoin *address* as the identity, the pubkeyhash better fits the definition of identity given in RFC003. Using a Bitcoin address as the identity would require implementations to do a number of cumbersome validation steps such as such as verifying that it is a p2pkh or p2wpkh address, extraction of the pubkeyhash, validation of the network.
+
+When using the pubkeyhash as one of the identities in an RFC003 message body it MUST be encoded as 20 hex encoded string.
+This RFC extends the [registry](./registry.md) with the following entry in the identity table:
+
+| Ledger   | Identity Name | JSON Encoding            | Description                                                                                  |
+|:----     |:-------       |:-------------            | -------------------------------------------------------------------------------------------- |
+| Bitcoin  | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA256 and then RIPEMD160 to a user's SECP256k1 compressed public key |
+
+## Hash Time Lock Contract
+
+### Hash Functions
+
+
+The `hash_function` header must be set to one of the following values if Bitcoin is used as a ledger:
+
+- SHA-256
+
+### Parameters
+
+The parameters for the Bitcoin HTLC follow [RFC003](./RFC-003-SWAP-basic.md) and are described concretely in the following table:
+
+| Variable        | Description                                                                                                               |
+|:----------------|:--------------------------------------------------------------------------------------------------------------------------|
+| asset           | The quantity of satoshi                                                                                                   |
+| secret          | The secret (32 bytes for SHA-256)                                                                                         |
```

Comment:
:question: Could it be described without using the word "secret"?

Reply:
Err I don't even know what it's doing there. Deleted!